crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1060,6 +1060,28 @@ async fn filter_payment_methods(
         let parse_result = serde_json::from_value::<PaymentMethodsEnabled>(payment_method);
         if let Ok(payment_methods_enabled) = parse_result {
             let payment_method = payment_methods_enabled.payment_method;
+            let allowed_payment_method_types = if let Some(payment_intent) = payment_intent {
```

Comment:
You can use `payment_intent.map()` instead of using `if let Some(...) = payment_intent { ... } else { None }`.